### PR TITLE
IGNITE-16040 Bump calcite version to 1.31.

### PR DESCRIPTION
**Pull request by zstan**
New calcite version contains fix for queries with several correlated sub-queries in select list.

### CODE REVIEW

File: modules/runner/src/integrationTest/sql/subquery/exists/test_correlated_exists.test_ignore [hunk 1]
```diff
@@ -13,9 +13,29 @@ PRAGMA enable_verification
...
+statement ok
+CREATE TABLE t1(a INTEGER, b INTEGER, c INTEGER, d INTEGER, e INTEGER, f INTEGER)
+
+statement ok
+INSERT INTO t1 VALUES (1, 2, 6, 4, 5, 6), (2, 3, 6, 5, 6, 7)
+
 statement ok
 INSERT INTO integers VALUES (1), (2), (3), (NULL)
 
+query IIIIII
```
Comment (korlov42) [inline modules/runner/src/integrationTest/sql/subquery/exists/test_correlated_exists.test_ignore:25]: we already have this very test in `select1.test` script

Comment (zstan) [reply modules/runner/src/integrationTest/sql/subquery/exists/test_correlated_exists.test_ignore:25]: in test1 returned resultset not checked, here i fill appropriate data and check result too.

Comment (korlov42) [reply modules/runner/src/integrationTest/sql/subquery/exists/test_correlated_exists.test_ignore:25]: then why do we have such a test at all?

File: modules/runner/src/integrationTest/sql/sqlite/join/join1.test_ignore [hunk 2]
```diff
@@ -0,0 +1,159 @@
+# Ignore https://issues.apache.org/jira/browse/CALCITE-5253
```
[the remark this answers is not in the document]
Comment (zstan) [reply modules/runner/src/integrationTest/sql/sqlite/join/join1.test_ignore:1]: I assure you that i not forget and i plan to complete with calcite issue soon

[the remark this answers is not in the document]
Comment (korlov42) [reply modules/runner/src/integrationTest/sql/sqlite/join/join1.test_ignore:1]: Well, patch in Calcite's upstream is not the only requirement. We need to bump a version in Ignite to acquire that fix as well. So, please file IGNITE ticket

[the remark this answers is not in the document]
Comment (zstan) [reply modules/runner/src/integrationTest/sql/sqlite/join/join1.test_ignore]: done

**Comment by zstan** [comment]
@korlov42 all done

**Review by korlov42 — APPROVED** [review]

**Pull request closed without merging**